src/AWS.Deploy.ServerMode.Client/CertificateVerificationEngine.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+
+namespace AWS.Deploy.ServerMode.Client
+{
+    public class CertificateVerificationEngine
+    {
+        private const string DEFAULT_DEPLOY_TOOL_ROOT = "dotnet aws";
+        private const string CERTIFICATE_SUBJECT_NAME = "Amazon Web Services, Inc.";
+
+        public virtual void VerifyCertificate(string deployToolRoot)
+        {
+            var deployToolDllPath = GetDeployToolDllPath(deployToolRoot);
+            var cert = new X509Certificate2(X509Certificate.CreateFromSignedFile(deployToolDllPath));
+            var chain = new X509Chain();
+            chain.ChainPolicy.RevocationMode = X509RevocationMode.Online;
+
+            // verifies that no certificate in the certificate chain is expired or revoked.
+            if (!chain.Build(cert))
+            {
+                throw new InvalidOperationException("The security certificate associated with AWS.Deploy.CLI.dll is either expired or revoked");
+            }
+            if (!string.Equals(cert.GetNameInfo(X509NameType.SimpleName, false), CERTIFICATE_SUBJECT_NAME, StringComparison.Ordinal))
```

Comment:
Is it enough to check the name? can't another certificate be created with the same name? don't we need to check the CA?